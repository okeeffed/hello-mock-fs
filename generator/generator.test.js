const mock = require("mock-fs");
const { generateFakerUtils } = require("./generator");
const path = require("path");
const fs = require("fs");

beforeEach(() => {
  mock({
    generator: {
      "example.ejs": mock.load(path.resolve(__dirname, "./example.ejs"), {
        lazy: false,
      }),
    },
    "fakerUtil.js": "",
    node_modules: mock.load(path.resolve(__dirname, "../node_modules")),
  });
});
afterEach(() => {
  mock.restore();
});

describe("generating a js file", () => {
  test("file output matches specific snapshot", () => {
    generateFakerUtils();
    const fileContents = fs.readFileSync(
      path.join(__dirname, "../fakerUtil.js"),
      "utf-8"
    );

    // ensure snapshot is not mocked
    mock.restore();

    expect(fileContents).toMatchSnapshot();
  });
});
