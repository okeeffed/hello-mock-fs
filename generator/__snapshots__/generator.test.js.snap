// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generating a js file file output matches specific snapshot 1`] = `
"const faker = require(\\"faker\\");

function getLoremWord() {
  return faker.lorem.word;
}

function getLoremWords() {
  return faker.lorem.words;
}

function getDatabaseColumn() {
  return faker.database.column;
}

module.exports = {
  getLoremWord,
  getLoremWords,
  getDatabaseColumn,
};
"
`;
